src/main/java/nextstep/security/authorization/web/RequestMatcherDelegatingAuthorizationManager.java

```diff
@@ -0,0 +1,30 @@
+package nextstep.security.authorization.web;
+
+import jakarta.servlet.http.HttpServletRequest;
+import java.util.List;
+import nextstep.security.authentication.Authentication;
+import nextstep.security.authorization.AuthorizationDecision;
+import nextstep.security.authorization.AuthorizationManager;
+import nextstep.security.util.RequestMatcher;
+import nextstep.security.util.RequestMatcherEntry;
+
+public class RequestMatcherDelegatingAuthorizationManager implements AuthorizationManager<HttpServletRequest> {
+
+    private static final AuthorizationDecision DENY = new AuthorizationDecision(false);
+
+    private final List<RequestMatcherEntry<AuthorizationManager>> mappings;
+
+    public RequestMatcherDelegatingAuthorizationManager(List<RequestMatcherEntry<AuthorizationManager>> mappings) {
+        this.mappings = mappings;
+    }
+
+    @Override
+    public AuthorizationDecision check(Authentication authentication, HttpServletRequest request) {
+        for (RequestMatcherEntry<AuthorizationManager> mapping : mappings) {
+            if (mapping.getRequestMatcher().matches(request)) {
```

Comment:
getter로 호출하여 일치 여부를 확인하는 것보다는 `mapping`에서 처리하도록 수정하면 책임이 명확하게 넘어갈 수 있겠네요 :)

Reply:
음.. 제 생각으로는 mapping인 RequestMatcherEntry는 RequestMatcher랑 AuthorizationManager를
단순히 묶어주는 역할을 수행한다고 생각해서, 
해당 객체에 `matches()`의 로직까지 있으면 오히려 책임이 과해지고,
원래 RequestMatcher의 역할이 흐려질것 같아 getter로 호출해서 일치 여부를 확인했습니다.

제가 제대로 이해한게 맞을까요? 잘못 이해한걸까요?

Reply:
```suggestion
            if (mapping.matches(request)) {
```
제가 의도 드렸던 내용은 getter를 꺼내와서 다시 호출하는 것이 아닌 해당 객체 내에서 처리할 수 있게 메시지를 던지는 형태를 말씀드린 것이었어요. getter를 호출해서 메소드 체이닝이 되는 것은 객체의 역할이 있다기보단 단순히 래핑클래스가 되어버려 책임과 역할이 불분명해질 수 있습니다.

```java
    public boolean matches(HttpServletRequest request) {
        return requestMatcher.matches(request);
    }
```